src/examples/logrotate

```diff
@@ -8,5 +8,6 @@
     delaycompress
     postrotate
         test -r /var/run/sssd.pid && /bin/kill -HUP `cat /var/run/sssd.pid 2>/dev/null` &>/dev/null || true
+        pgrep -c sssd_kcm &>/dev/null && /bin/kill -HUP `pgrep sssd_kcm 2>/dev/null` &>/dev/null || true
```

Comment:
hmm, pgrep is not in our dependency (yet). I tried in the container, installed sssd and pgrep is not there (/bin/kill is).

Reply:
I wonder if we can use `systemctl reload` somehow because systemd should know the pid.
Can we define 

    ExecReload=/bin/kill -HUP $MAINPID

in the systemd unit and use it in logrotate?

Maybe to unify this with SSSD service too...WDYT

Reply:
That would be a very simple solution. However, I think it requires some design discussion within the team as `systemctl reload` is expected to reload the configuration, not to rotate the logs.

`pgrep`  and `pkill` are tools that are present _most_ of the time, so it is not that dangerous to consider them to be present (although they could be missing, I have to admit it). An extra detail, we are not the only ones using them after log rotation.

It think setting them as a very weak dependency (keyword `Suggests`) is a good compromise. I would have preferred a _weak_ dependency (keyword `Recommends`) instead of _very weak_, but SSSD has a _very weak_ dependency on `logrotate`.

And if we really want to avoid all this, we could create a PID file.

Reply:
On the other hand, we could fall back to more _standard_ commands such as `ps` and `awk`, which have a very high probability of being present in the system, and use this line to retrieve the PID:

`ps --ppid 1 --format pid=,cmd= | awk '/sssd_kcm / { print $1 }'`

This line has the advantage of providing the one instance of KCM that was launched by `systemd` (the `--ppid 1` filter) which is a good way to handle containers (a process launched by `systemd` will have 1 as parent when seen from inside a container but not from the outside). On the contrary, this will not work if KCM was launched by any other means (not using `systemd` at all). Should we worry about this case? If someone decided to manually launch KCM, they should also take care of the logs rotation.

More important, `ps`, `pgrep` and `pkill` are all provided by the same package: `procps-ng`. So, if we have `ps`, we also have `pkill`.

`awk` is provided by its own separate package, which adds a new (probably _very weak_ dependency.

I think we should keep things simple and just consider `pkill` is present, while maintaining the _very weak_ dependency on the package `procps-ng` which costs us nothing.

BTW, I just noticed `pkill` has the `--parent` parameters which works the same as the `--ppid` filter and we can use it to correctly handle the containers. But again, this will not work if `sssd_kcm` was manually launched.

Reply:
> a process launched by `systemd` will have 1 as parent when seen from inside a container but not from the outside

JFTR: I don't think it's a common approach to use systemd to manage processes within a container at all.

Reply:
Good point. No `--parent 1` then.